lib/ansible/modules/cloud/vmware/vmware_serial_port.py

```diff
@@ -0,0 +1,345 @@
+
+        Return:
+          - result object
+
+        """
+        spec = vim.vm.ConfigSpec()
+        spec.deviceChange = []
+        serial_spec = vim.vm.device.VirtualDeviceSpec()
+        serial_spec.operation = 'add'
+        serial_port = vim.vm.device.VirtualSerialPort()
+        serial_port.yieldOnPoll = True
+        serial_port.backing = self.get_backing_info(serial_port, self.params.get('backing'))
+        serial_spec.device = serial_port
+        spec.deviceChange.append(serial_spec)
+        task = vm.ReconfigVM_Task(spec=spec)
+        self.wait_for_task(task)
+        if task.info.state == 'error':
+            return {'changed': self.change_applied, 'failed': True, 'msg': task.info.error.msg}
+        else:
+            return {'changed': self.change_applied, 'failed': False, 'msg': "Serial port successfully created"}
+
+    def wait_for_task(self, task, poll_interval=1):
+        """
+        Wait for a VMware task to complete.  Terminal states are 'error' and 'success'.
+
+        Inputs:
+          - task: the task to wait for
+          - poll_interval: polling interval to check the task, in seconds
+
+        Modifies:
+          - self.change_applied
+        """
+        while task.info.state not in ['error', 'success']:
+            time.sleep(poll_interval)
+        self.change_applied = self.change_applied or task.info.state == 'success'
+
+    def get_backing_info(self, serial_port, backing):
+        switcher = {
+            "network": self.set_network_backing,
+            "pipe": self.set_pipe_backing,
+            "device": self.set_device_backing,
+            "file": self.set_file_backing
+        }
+        backing_func = switcher.get(backing, "Invalid Backing Info")
+        return backing_func(serial_port)
+
+    def set_network_backing(self, serial_port):
+        backing = serial_port.URIBackingInfo()
+        backing.serviceURI = self.params.get('service_uri')
+        backing.direction = self.params.get('direction')
+        backing.proxyURI = self.params.get('proxy_uri')
+        return backing
+
+    def set_pipe_backing(self, serial_port):
+        backing = serial_port.PipeBackingInfo()
+        backing.pipeName = self.params.get('pipe_name')
+        backing.endpoint = self.params.get('endpoint')
+        backing.noRxLoss = self.params.get('no_rx_loss')
+        return backing
+
+    def set_device_backing(self, serial_port):
+        backing = serial_port.DeviceBackingInfo()
+        backing.deviceName = self.params.get('device_name')
+        return backing
+
+    def set_file_backing(self, serial_port):
+        backing = serial_port.FileBackingInfo()
+        backing.fileName = self.params.get('file_path')
+        return backing
+
+
+def main():
+    argument_spec = vmware_argument_spec()
+    argument_spec.update(
+        uuid=dict(type='str'),
+        vm_id=dict(type='str'),
+        backing=dict(type='str', choices=['network', 'file', 'pipe', 'device']),
```

Comment:
Could you please consider handling multiple serial port ? Please take a look at `vmware_guest_network` module for such functionality where it handles multiple network devices at single time.